### ISSUE TITLE
open() doesn't seem to honor lazy_load and/or copy_arrays

### ISSUE THREAD

**Issue by eigenbrot** [asdf-format/asdf#867]
Somewhere between version 2.4.2 and 2.7.0 the following code stopped working:

```
import numpy as np
import asdf
x = np.arange(100)
d = {'TEST': {'data': x}}
asdf.AsdfFile(d).write_to('tmp.asdf')
with asdf.open('tmp.asdf', lazy_load=False, copy_arrays=True) as f:
 d = f.tree

d['TEST']['data'][0]
```

In 2.4.2 the result was as expected but in 2.7.0 the result is:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/XXX/.conda/envs/test/lib/python3.8/site-packages/asdf/tags/core/ndarray.py", line 524, in __operation__
    return getattr(self._make_array(), name)(*args)
  File "/home/XXX/.conda/envs/test/lib/python3.8/site-packages/asdf/tags/core/ndarray.py", line 256, in _make_array
    if block_data is not self.block.data:
  File "/home/XXX/.conda/envs/test/lib/python3.8/site-packages/asdf/block.py", line 1169, in data
    raise IOError(
OSError: ASDF file has already been closed. Can not get the data.
```
I thought ``lazy_load`` and ``copy_arrays`` were supposed to stop exactly this behavior (and they did back in 2.4.2).

Any insight is appreciated. Thanks!

**Comment by eslavich** [comment]
Hi, thanks for the report.  That's definitely a bug, and this PR is the culprit:

https://github.com/asdf-format/asdf/pull/716

We'll want to make a 2.7.1 release to fix this, I think.

**Comment by eigenbrot** [comment]
Thanks for fixing this so fast, @eslavich!

**Comment by eslavich** [comment]
No problem!  The fix is in 2.7.1, which has been released to pypi.
